system.go

```diff
@@ -78,8 +98,12 @@ func Processes() ([]types.Process, error) {
 // Self return a types.Process object representing this process. If process
 // information collection is not implemented for this platform then
 // types.ErrNotImplemented is returned.
-func Self() (types.Process, error) {
-	provider := registry.GetProcessProvider()
+func Self(opts ...ProviderOption) (types.Process, error) {
+	options := registry.ProviderOptions{}
+	for _, opt := range opts {
+		opt(&options)
+	}
+	provider := registry.GetProcessProvider(options)
```

Comment:
nit: maybe could extract this 5 liner as a func, since it's repeated in 4 places.